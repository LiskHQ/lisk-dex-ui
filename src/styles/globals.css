:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}