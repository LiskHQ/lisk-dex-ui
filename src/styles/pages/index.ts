import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}