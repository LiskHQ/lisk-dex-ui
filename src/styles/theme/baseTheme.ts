export const baseTheme = {
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 600,
    },
    h3: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 600,
    },
    h5: {
      fontSize: "1rem",
      lineHeight: "1.25rem",
      fontWeight: 500,
    },
    h6: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    body1: {
      fontSize: "1rem",
      lineHeight: "1.375rem",
      fontWeight: 500,
    },
    body2: {
      fontSize: "0.875rem",
      lineHeight: "1.125rem",
      fontWeight: 400,
    },
    caption: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    subtitle1: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    subtitle2: {
      fontSize: "1.25rem",
      lineHeight: "1.5rem",
      fontWeight: 400,
    }
  },
  icons: {
    sm: "1rem",
    md: "1.25rem",
    lg: "1.5rem",
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
    },
  },
  shadows: {
    sm: "0px 1px 3px rgba(0, 0, 0, 0.1)"
  }
}