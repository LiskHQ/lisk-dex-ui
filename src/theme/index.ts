export { default as theme } from "./theme";
export { default as variables } from "./variables";
