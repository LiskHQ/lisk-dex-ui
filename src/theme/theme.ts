import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
