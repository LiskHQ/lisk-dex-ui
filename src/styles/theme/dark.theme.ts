import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#F2F2F2',
    secondary: '#F2F2F2',
    paragraph: '#CDCDCD',
    body: '#F2F2F2',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#A4ABB7',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#5845CC',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#E6E6E6',
  },
  secondary: {
    0: '#F5981B'
  },
  primary: {
    85: '#7E6CF4',
    70: '#4F39B2',
    60: '#271B5B',
    40: '#2D2166',
    20: '#322573',
    10: '#4F39B2',
    5: '#322573',
    2.5: '#25195E',
    1: '#F4F8F9',
    0: '#FFFFFF',
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#E6E6E6',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#433199',
  },
  bg: {
    primary: 'linear-gradient(90deg, #140C42 0%, #211953 52.88%, #110A3A 101.53%, #140C42 101.53%)',
    secondary: '#22184C',
    helper: '#6953F4',
    proposalItem: 'linear-gradient(180deg, #2E2567 0%, rgba(37, 26, 103, 0.5) 143.01%);',
    banner: 'linear-gradient(180deg, #150556 0%, rgba(25, 9, 98, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #372589 0%, rgba(27, 19, 68, 0.8) 100%)',
    slider: '#322573',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#2EA3DD',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'dark',
  },
})

export default lightTheme
