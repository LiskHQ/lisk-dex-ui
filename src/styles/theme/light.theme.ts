import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#282828',
    secondary: '#515661',
    paragraph: '#6B7280',
    body: '#6B7280',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#6B7280',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#A196F6',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#161133',
  },
  primary: {
    85: '#7E6CF4',
    60: '#A196F6',
    40: '#E4EBF9',
    20: '#D9D8F8',
    10: '#E7E9F9',
    5: '#EEF0F8',
    2.5: '#F2F5F9',
    1: '#F4F8F9',
    0: '#6953F4'
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#535353',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#E6E6E6',
  },
  bg: {
    primary: '#FBFBFB',
    secondary: '#FFFFFF',
    helper: '#22184D',
    banner: 'linear-gradient(180deg, #F3F3FD 0%, rgba(238, 238, 249, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #EEEFF7 0%, rgba(232, 234, 245, 0.8) 100%);',
    slider: '#F2F2F2',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#EBF7F6',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'light'
  },
})

export default lightTheme
