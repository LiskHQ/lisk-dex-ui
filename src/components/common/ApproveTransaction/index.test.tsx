import { ThemeProvider } from "@mui/material";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { lightTheme } from "styles/theme";
import { ApproveTransactionModal, IApproveTransactionModalProps } from ".";

function renderComponent(props: IApproveTransactionModalProps) {
  return render(
    <ThemeProvider theme={lightTheme}>
      <ApproveTransactionModal {...props} />
    </ThemeProvider>
  );
}

describe("Button component", () => {
  const mockProps: IApproveTransactionModalProps = {
    approvingTransaction: false,
    onClose: jest.fn(),
    onConfirm: jest.fn(),
  }

  it("checks if the component matches the snapshot", () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it("checks all buttons click work fine", () => {
    const { getByText } = renderComponent(mockProps);

    fireEvent.click(getByText("Cancel"));
    expect(mockProps.onClose).toBeCalled();

    fireEvent.click(getByText("Send wallet request"));
    expect(mockProps.onConfirm).toBeCalled();
  });

});