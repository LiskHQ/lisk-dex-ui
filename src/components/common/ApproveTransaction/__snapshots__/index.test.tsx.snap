// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component checks if the component matches the snapshot 1`] = `
.emotion-0 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 201;
}

.emotion-0 .approve-transaction-background {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  z-index: 200;
}

.emotion-0 .approve-transaction-modal-container {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  float: right;
  margin-bottom: 10rem;
  width: 34.45rem;
  border-radius: 0.25rem;
  background: #FBFBFB;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1);
  z-index: 201;
}

@media (max-width:767.95px) {
  .emotion-0 .approve-transaction-modal-container {
    width: 100%;
    top: 5.5rem;
  }
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1.5rem 1rem 1.5rem 1rem;
  color: #282828;
  border-bottom: 0.5px solid #E6E6E6;
}

@media (max-width:767.95px) {
  .emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header {
    padding: 1rem;
  }
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header svg {
  cursor: pointer;
  margin-left: 0.75rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header .approve-transaction-status {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header .approve-transaction-status .transaction-status {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.5rem 1rem 0.5rem 1rem;
  border: 0.5px solid #E6E6E6;
  border-radius: 0.75rem;
  background: #EEF0F8;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-header .approve-transaction-status .transaction-status .status {
  content: " ";
  border-radius: 100%;
  width: 0.625rem;
  height: 0.625rem;
  background: #8DC881;
  margin-right: 0.5rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body {
  padding: 1.5rem 1.5rem 1rem 1.5rem;
}

@media (max-width:767.95px) {
  .emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body {
    padding: 1.5rem 1rem 1rem 1rem;
  }
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 2rem;
}

@media (max-width:767.95px) {
  .emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info {
    margin-top: 1.5rem;
  }
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .MuiFormLabel-root {
  color: #282828;
  margin-bottom: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.124rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .approve-transaction-account {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .approve-transaction-account .approve-transaction-account-address {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .approve-transaction-balance {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .approve-transaction-balance .approve-transaction-balance-amount {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-wallet-info .approve-transaction-balance .approve-transaction-balance-amount svg {
  width: 1.25rem;
  height: 1.25rem;
  margin-left: 0.25rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-estimated-balance-change {
  margin-top: 2rem;
}

@media (max-width:767.95px) {
  .emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-estimated-balance-change {
    display: none;
  }
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-estimated-balance-change .approve-transaction-estimated-balance {
  margin-top: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-estimated-balance-change .approve-transaction-estimated-balance .estimated-balance {
  margin-left: 1rem;
  color: #FF4557;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-request {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  width: 100%;
  padding: 6rem 0 6rem 0;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-request svg {
  width: 4rem;
  height: 4rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-request h4 {
  margin-top: 0.75rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-request p {
  margin-top: 0.5rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-proposal-creation-fee {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 1.5rem;
  margin-bottom: 0.75rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-body .approve-transaction-proposal-transaction-total {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-top: 0.5px solid #E6E6E6;
  padding: 1rem 0 0 0;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-footer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.5rem 2.5rem 2.5rem 1.5rem;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-footer .approve-transaction-modal-cancel {
  width: 10rem;
  margin-right: 1rem;
  background: none;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-footer .approve-transaction-modal-cancel:hover P {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #FF4557;
}

.emotion-0 .approve-transaction-modal-container .approve-transaction-modal-footer .approve-transaction-modal-cancel P {
  color: #FF4557;
}

.emotion-4 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-8 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.25rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0em;
}

.emotion-10 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-13 {
  color: rgba(0, 0, 0, 0.6);
  font-size: 1rem;
  line-height: 1.4375em;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
}

.emotion-13.Mui-focused {
  color: #1976d2;
}

.emotion-13.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-13.Mui-error {
  color: #d32f2f;
}

.emotion-28 {
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-33 {
  margin: 0;
  font-size: 1.15rem;
  line-height: 1.25rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00714em;
}

.emotion-36 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
  width: 100%;
  background: #6953F4;
  color: #FFFFFF;
  padding: 0.75rem;
  text-transform: capitalize;
}

.emotion-36::-moz-focus-inner {
  border-style: none;
}

.emotion-36.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-36 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-36:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-36:hover {
    background-color: transparent;
  }
}

.emotion-36.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-36:hover {
  background: #A196F6;
}

.emotion-36.Mui-disabled {
  color: #FFFFFF;
  opacity: 0.7;
}

.emotion-38 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="approve-transaction-background MuiBox-root emotion-1"
    />
    <div
      class="approve-transaction-modal-container MuiBox-root emotion-1"
    >
      <div
        class="approve-transaction-modal-header MuiBox-root emotion-1"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 emotion-4"
        >
          Approve transaction
        </h4>
        <div
          class="approve-transaction-status MuiBox-root emotion-1"
        >
          <div
            class="transaction-status MuiBox-root emotion-1"
          >
            <div
              class="status MuiBox-root emotion-1"
            />
            <h5
              class="MuiTypography-root MuiTypography-h5 emotion-8"
            >
              Lisk Testnet
            </h5>
          </div>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_625_48983)"
            >
              <path
                d="M15 5L5 15"
                stroke="#282828"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.66667"
              />
              <path
                d="M5 5L15 15"
                stroke="#282828"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.66667"
              />
            </g>
            <defs>
              <clippath
                id="clip0_625_48983"
              >
                <rect
                  fill="white"
                  height="20"
                  width="20"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
      <div
        class="approve-transaction-modal-body MuiBox-root emotion-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-10"
        >
          This action will approve this transaction.
        </p>
        <div
          class="approve-transaction-wallet-info MuiBox-root emotion-1"
        >
          <div
            class="approve-transaction-account MuiBox-root emotion-1"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary emotion-13"
            >
              Account:
            </label>
            <div
              class="approve-transaction-account-address MuiBox-root emotion-1"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-10"
              >
                0x90F8...c9C1
              </p>
            </div>
          </div>
          <div
            class="approve-transaction-balance MuiBox-root emotion-1"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary emotion-13"
            >
              Balance:
            </label>
            <div
              class="approve-transaction-balance-amount MuiBox-root emotion-1"
            >
              <svg
                fill="#6953F4"
                height="48"
                viewBox="0 0 48 48"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M20.7155 12.7156L23.437 8.07038C23.4839 7.97654 23.6012 7.97654 23.6246 8.07038L37.0675 31.0617C37.091 31.1086 37.0675 31.1555 37.0441 31.2024L29.3021 39.9063C29.2786 39.9297 29.2317 39.9532 29.2082 39.9532H22.9208C22.8035 39.9532 22.7566 39.8359 22.827 39.7655L27.0264 35.0265L30.9678 30.6159C31.0147 30.569 31.0147 30.5221 30.9912 30.4752L23.5308 17.6657L20.7155 12.8329C20.6921 12.8094 20.6921 12.7625 20.7155 12.7156ZM24.7508 35.238L20.5748 39.9301C20.5631 39.9419 20.5455 39.9536 20.5279 39.9653C20.5103 39.977 20.4927 39.9888 20.481 40.0005H17.8534C17.8065 40.0005 17.783 39.977 17.7596 39.9536L10.0176 31.2497C9.99413 31.2028 9.99413 31.1559 10.0176 31.109L19.6599 14.5459C19.7068 14.4755 19.8241 14.4755 19.871 14.5459L22.569 19.191C22.5924 19.2145 22.5924 19.2614 22.569 19.3083L16.0469 30.499C16.0235 30.5459 16.0469 30.5928 16.0704 30.6398L19.9648 35.0034C19.9883 35.0269 20.0352 35.0503 20.0587 35.0503H24.6569C24.7742 35.0503 24.8212 35.1676 24.7508 35.238Z"
                  fill-rule="evenodd"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-10"
              >
                145,321 LSKDEX (~$136,461)
              </p>
            </div>
          </div>
        </div>
        <div
          class="approve-transaction-estimated-balance-change MuiBox-root emotion-1"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 emotion-8"
          >
            Estimated balance change
          </h5>
          <div
            class="approve-transaction-estimated-balance MuiBox-root emotion-1"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 emotion-8"
            >
              LSK:
            </h5>
            <h5
              class="MuiTypography-root MuiTypography-h5 estimated-balance emotion-8"
            >
              -5000 LSK
            </h5>
          </div>
        </div>
        <div
          class="approve-transaction-request MuiBox-root emotion-1"
        >
          <svg
            fill="#6953F4"
            height="48"
            viewBox="0 0 48 48"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M20.7155 12.7156L23.437 8.07038C23.4839 7.97654 23.6012 7.97654 23.6246 8.07038L37.0675 31.0617C37.091 31.1086 37.0675 31.1555 37.0441 31.2024L29.3021 39.9063C29.2786 39.9297 29.2317 39.9532 29.2082 39.9532H22.9208C22.8035 39.9532 22.7566 39.8359 22.827 39.7655L27.0264 35.0265L30.9678 30.6159C31.0147 30.569 31.0147 30.5221 30.9912 30.4752L23.5308 17.6657L20.7155 12.8329C20.6921 12.8094 20.6921 12.7625 20.7155 12.7156ZM24.7508 35.238L20.5748 39.9301C20.5631 39.9419 20.5455 39.9536 20.5279 39.9653C20.5103 39.977 20.4927 39.9888 20.481 40.0005H17.8534C17.8065 40.0005 17.783 39.977 17.7596 39.9536L10.0176 31.2497C9.99413 31.2028 9.99413 31.1559 10.0176 31.109L19.6599 14.5459C19.7068 14.4755 19.8241 14.4755 19.871 14.5459L22.569 19.191C22.5924 19.2145 22.5924 19.2614 22.569 19.3083L16.0469 30.499C16.0235 30.5459 16.0469 30.5928 16.0704 30.6398L19.9648 35.0034C19.9883 35.0269 20.0352 35.0503 20.0587 35.0503H24.6569C24.7742 35.0503 24.8212 35.1676 24.7508 35.238Z"
              fill-rule="evenodd"
            />
          </svg>
          <h4
            class="MuiTypography-root MuiTypography-h4 emotion-4"
          >
            Approve request
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            https://liskdex.io
          </p>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 emotion-28"
        >
          Transaction Summary
        </h6>
        <div
          class="approve-transaction-proposal-creation-fee MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Proposal creation fee:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            5000 LSKDEX (~$4952.42)
          </p>
        </div>
        <div
          class="approve-transaction-proposal-transaction-total MuiBox-root emotion-1"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 emotion-33"
          >
            Transaction total:
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 emotion-33"
          >
            5000 LSKDEX (~$4952.42)
          </h6>
        </div>
      </div>
      <div
        class="approve-transaction-modal-footer MuiBox-root emotion-1"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root approve-transaction-modal-cancel emotion-36"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Cancel
          </p>
          <span
            class="MuiTouchRipple-root emotion-38"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root approve-transaction-modal-confirm emotion-36"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-10"
          >
            Send wallet request
          </p>
          <span
            class="MuiTouchRipple-root emotion-38"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
